### PR TITLE
Add event-level report validator

### PR DESCRIPTION
**Pull request by apasel422**
This will be used as the basis for an independent interop-test-output checker.

### CODE REVIEW

File: ts/src/header-validator/validate-json.ts [hunk 1]
```diff
@@ -362,26 +370,41 @@ function suitableScope(
 
   const scoped = scope(url)
   if (s !== scoped) {
+    if (rejectExtraComponents) {
```
Comment (linnan-github) [inline ts/src/header-validator/validate-json.ts:373]: Shall we add full test coverage?

Comment (apasel422) [reply ts/src/header-validator/validate-json.ts:373]: Done.

File: ts/src/header-validator/validate-json.ts [hunk 2]
```diff
@@ -1542,3 +1565,112 @@ export function validateTrigger(
...
+}
+
+function randomUuid(ctx: Context, j: Json): Maybe<string> {
+  return string(ctx, j).filter((s) => {
+    if (!uuid.validate(s)) {
+      ctx.error('must be a valid UUID')
+      return false
+    }
+    if (uuid.version(s) !== 4) {
+      ctx.error('must be a version 4 (random) UUID')
```
Comment (linnan-github) [inline ts/src/header-validator/validate-json.ts:1604]: Please add test coverage.

File: ts/src/header-validator/validate-json.ts [hunk 3]
```diff
@@ -1542,3 +1565,112 @@ export function validateTrigger(
...
+}
+
+function reportDestination(ctx: Context, j: Json): Maybe<string | string[]> {
+  const suitableSiteNoExtraneous = (ctx: Context, j: Json) =>
+    suitableSite(ctx, j, /*rejectExtraComponents=*/ true)
+
+  return typeSwitch<string | string[]>(ctx, j, {
+    string: (ctx, j) => suitableSiteNoExtraneous(ctx, j),
+    list: (ctx, j) =>
+      array(ctx, j, suitableSiteNoExtraneous, { minLength: 1, maxLength: 3 }),
```
Comment (linnan-github) [inline ts/src/header-validator/validate-json.ts:1589]: Please add test coverage.

File: ts/src/header-validator/validate-json.ts [hunk 4]
```diff
@@ -1542,3 +1565,112 @@ export function validateTrigger(
...
+  const endpoint = (ctx: Context, j: Json): Maybe<number> =>
+    number(ctx, j)
+      .filter((n) => isInteger(ctx, n))
+      .filter((n) =>
+        isInRange(
+          ctx,
+          n,
+          prev,
+          UINT32_MAX,
+          `must be >= ${prevDesc} (${prev}) and <= uint32 max (${UINT32_MAX})`
```
Comment (linnan-github) [inline ts/src/header-validator/validate-json.ts:1624]: Please add test coverage.

File: ts/src/header-validator/validate-json.ts [hunk 5]
```diff
@@ -1542,3 +1565,112 @@ export function validateTrigger(
...
+      attributionDestination: field(
+        'attribution_destination',
+        reportDestination
+      ),
+      randomizedTriggerRate: field(
+        'randomized_trigger_rate',
+        randomizedTriggerRate
+      ),
+      reportId: field('report_id', randomUuid),
+      scheduledReportTime: field('scheduled_report_time', int64),
```
Comment (linnan-github) [inline ts/src/header-validator/validate-json.ts:1654]: Do we need test overage for types of these fields?

File: ts/src/header-validator/validate-json.ts (hunk 4 above)
Comment (apasel422) [reply ts/src/header-validator/validate-json.ts:1624]: Which part doesn't have coverage?

Comment (linnan-github) [reply ts/src/header-validator/validate-json.ts:1624]: Sorry, this one is fine.

File: ts/src/header-validator/validate-json.ts (hunk 5 above)
Comment (apasel422) [reply ts/src/header-validator/validate-json.ts:1654]: Done.

File: ts/src/header-validator/validate-json.ts (hunk 2 above)
Comment (apasel422) [reply ts/src/header-validator/validate-json.ts:1604]: Done.

File: ts/src/header-validator/validate-json.ts (hunk 3 above)
Comment (apasel422) [reply ts/src/header-validator/validate-json.ts]: Done.

**Review by linnan-github — APPROVED** [review]

**Pull request merged**